<launch>
  <arg name="map_suffix"/>

	<!-- 2d map -->
	<node pkg="map_server" name="map2d" type="map_server"
		  args="$(find gazebo_simulation)/maps/map-$(arg map_suffix).yaml" respawn="true" >
		<remap from="map" to="map_server_map"/>
	</node>

	<!-- 3d map -->
	<node machine="c2" name="octomap_server_combined" pkg="octomap_server" 
			type="octomap_server_node" args="$(find gazebo_simulation)/maps/map-$(arg map_suffix).bt" >
		<param name="resolution" value="0.025" />
		<param name="sensor_model/range" value="4.0" />

		<param name="sensor_model/hit" value= "0.8" />
		<param name="sensor_model/miss" value= "0.31" />
		<param name="sensor_model/min" value= "0.12" />
		<param name="sensor_model/max" value= "0.95" />

		<param name="min_z_range" value="0.05" />
		<param name="max_z_range" value="1.8" />
		<param name="min_x_size" value="25.0" />
		<param name="min_y_size" value="25.0" />

		<param name="pointcloud_min_z" value="0.10"/>
		<param name="pointcloud_max_z" value="1.80"/>
		<param name="filter_ground" value="true"/>
		<param name="occupancy_min_z" value="0.05"/>
		<param name="occupancy_max_z" value="2.0"/>


		<!-- <remap from="cloud_in" to="/narrow_stereo_textured/points_filtered_attached2" /> -->
		<remap from="cloud_in" to="/narrow_stereo_textured/points2_filtered" />
	</node>

	<!-- localization -->
	<node pkg="amcl" type="amcl" name="amcl" output="screen">
		<remap from="scan" to="base_scan" />
		<!-- Publish scans from best pose at a max of 10 Hz -->
		<param name="base_frame_id" value="base_footprint"/>
		<param name="odom_model_type" value="omni"/>
		<param name="odom_alpha5" value="0.1"/>
		<param name="transform_tolerance" value="0.2" />
		<param name="gui_publish_rate" value="10.0"/>
		<param name="laser_max_beams" value="30"/>
		<param name="min_particles" value="500"/>
		<param name="max_particles" value="5000"/>
		<param name="kld_err" value="0.05"/>
		<param name="kld_z" value="0.99"/>
		<param name="odom_alpha1" value="0.2"/>
		<param name="odom_alpha2" value="0.2"/>
		<!-- translation std dev, m -->
		<param name="odom_alpha3" value="0.8"/>
		<param name="odom_alpha4" value="0.2"/>
		<param name="laser_z_hit" value="0.5"/>
		<param name="laser_z_short" value="0.05"/>
		<param name="laser_z_max" value="0.05"/>
		<param name="laser_z_rand" value="0.5"/>
		<param name="laser_sigma_hit" value="0.2"/>
		<param name="laser_lambda_short" value="0.1"/>
		<param name="laser_lambda_short" value="0.1"/>
		<param name="laser_model_type" value="likelihood_field"/>
		<!-- <param name="laser_model_type" value="beam"/> -->
		<param name="laser_likelihood_max_dist" value="2.0"/>
		<param name="update_min_d" value="0.2"/>
		<param name="update_min_a" value="0.5"/>
		<param name="odom_frame_id" value="odom_combined"/>
		<param name="resample_interval" value="1"/>
		<param name="transform_tolerance" value="0.1"/>
		<param name="recovery_alpha_slow" value="0.0"/>
		<param name="recovery_alpha_fast" value="0.0"/>
		<remap from="map" to="map_server_map"/>
	</node>

	<!-- service that re-publishes AMCL poses so that a recent pose is available whenever -->
	<node pkg="pr2_plan_utils" name="posegossip" type="posegossip.py"/>


</launch>
